### ISSUE TITLE
Global optimization of surface adsorbates. 

### ISSUE THREAD

**Issue by jungsdao** [libAtoms/workflow#263]
Hello, 
In my [previous paper](https://www.nature.com/articles/s41524-023-01065-w), I was using iterative GAP training workflow in my own script.
I wanted to incorporate this workflow into wfl package and I think now it works quite well within wfl package. 
I guess some of people might find this functionality useful. 
I'm not sure how I could contribute in this regard, any advice would be appreciated!

**Comment by bernstei** [comment]
I've been thinking about this, and the fact that the current re-implementation of the iterative fit from the 2019 de novo exploration is inside wfl (in cli.py).  I've come to think that the top-level scripts probably don't belong "inside" wfl, because they _use_ wfl, but aren't really part of it (although your modifications to the minima hopping probably do belong inside it).

I really want to refactor my `gap_rss_iter_fit` anyway, so maybe this is the time to start. You should think about what of your changes are low level, in some sense (i.e. might be reusable by a wide range of people/tasks), and what's a particular high-level combination of operations (i.e. separate workflow operations from the actual top-level workflow). The former we can add to something inside wfl.  The top level stuff I suggest we make a "scripts" directory, or something like that, and leave the `wfl/cli/` for simple wrappers of individual wfl functions.

**Comment by bernstei** [comment]
Yes - let's start with a PR for the low level changes, and we'll see how much (if any) that breaks. Then we can think about the best place to put the top-level script, if you want to contribute that as well.
